cpp/cmake/common.cmake

```diff
@@ -0,0 +1,95 @@
+cmake_minimum_required(VERSION 3.16)
+
+if (WIN32)
+# Temporary see https://github.com/zeroc-ice/ice/issues/3476
+set(CMAKE_CXX_STANDARD 20)
+else()
+set(CMAKE_CXX_STANDARD 17)
+endif()
+
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+find_package(Threads REQUIRED)
+
+if (NOT Ice_HOME)
```

Comment:
Yes, there is lots of cleanup and fixes to do here to support various settings and configurations.

Reply:
Did you try using FindIce CMake module?

https://github.com/Kitware/CMake/blob/master/Modules/FindIce.cmake

Reply:
I ask because users will end up copying our `common.cmake` when it would be better for most users to use the FindIce.cmake module.

Reply:
Once we release hopefully we can switch to using `find_package(Ice)` but for now we'll have to use ours.